src/main/java/flight/reservation/flight/Schedule.java

```diff
@@ -23,16 +23,20 @@ public void scheduleFlight(Flight flight, Date date) {
     }
 
     public void removeFlight(Flight flight) {
-        List<ScheduledFlight> tbr = new ArrayList<>();
+        List<ScheduledFlight> toBeRemovedFlights = new ArrayList<>();
         for (ScheduledFlight scheduledFlight : scheduledFlights) {
-            if (scheduledFlight == flight ||
-                    (flight.getArrival() == scheduledFlight.getArrival() &&
-                            flight.getDeparture() == scheduledFlight.getDeparture() &&
-                            flight.getNumber() == scheduledFlight.getNumber())) {
-                tbr.add(scheduledFlight);
+            if (checkIsToBeRemoved(flight, scheduledFlight)) {
+                toBeRemovedFlights.add(scheduledFlight);
             }
         }
-        scheduledFlights.removeAll(tbr);
+        scheduledFlights.removeAll(toBeRemovedFlights);
+    }
+
+    private boolean checkIsToBeRemoved(Flight flight, ScheduledFlight scheduledFlight) {
+       return (scheduledFlight == flight ||
+               (flight.getArrival() == scheduledFlight.getArrival() &&
+                       flight.getDeparture() == scheduledFlight.getDeparture() &&
+                       flight.getNumber() == scheduledFlight.getNumber()));
```

Comment:
```suggestion
       return ((flight.getArrival() == scheduledFlight.getArrival() &&
                       flight.getDeparture() == scheduledFlight.getDeparture() &&
                       flight.getNumber() == scheduledFlight.getNumber()));
```
Comparing the reference is currently confusing.

If you want this for performance reasons, try to make this "escalation strategy" more clear with dedicated function calls: 
```
return isSameObject(scheduledFlight, flight) || hasSameProperties(scheduledFlight, flight)
```